frontend/packages/dev-console/src/components/formik-fields/field-types.ts

```diff
@@ -1,30 +1,53 @@
-export interface InputFieldProps {
+import { TextInputTypes } from '@patternfly/react-core';
+
+export interface FieldProps {
   name: string;
-  type?: string;
   label?: string;
   helpText?: string;
   required?: boolean;
+}
+
+export interface InputFieldProps extends FieldProps {
+  type: TextInputTypes;
   placeholder?: string;
   onChange?: (event) => void;
   onBlur?: (event) => void;
 }
 
+export interface CheckboxFieldProps extends FieldProps {
+  formLabel?: string;
+}
+
 export interface SearchInputFieldProps extends InputFieldProps {
   onSearch: (searchTerm: string) => void;
 }
 
-export interface DropdownFieldProps extends InputFieldProps {
+export interface DropdownFieldProps extends FieldProps {
   items?: object;
   selectedKey?: string;
   title?: React.ReactNode;
   fullWidth?: boolean;
 }
 
-export interface EnvironmentFieldProps extends InputFieldProps {
+export interface EnvironmentFieldProps extends FieldProps {
   obj?: object;
   envPath: string[];
 }
 
+export interface ResourceLimitFieldProps extends FieldProps {
+  unitName: string;
+  dropdownUnits: object;
```

Comment:
You mean type of `dropdownUnits` or name?

Reply:
At first i didn't like the name, but saw afterwards that it's coming from an existing component in console. I do not like naming things according to their UI representation because if we ever changed it, the prop wouldn't make much sense then. If we could, I'd probably have named it `unitOptions`.

But this particular comment was referring to the type being `object`.

Reply:
I've updated the prop names. But the type of the prop needs to be an object since the dropdown can have a generic object or enum as its options.